src/util/listQueryBuilder.js

```diff
@@ -623,14 +623,13 @@ function getMinRangeValue (rangeOrValue) {
   const isSingleValue = !_.isPlainObject(rangeOrValue)
   if (isSingleValue) return rangeOrValue
 
-  const rangeValues = [rangeOrValue.gt, rangeOrValue.gte]
-  const [minValue] = rangeValues.sort((a, b) => {
-    if (_.isUndefined(a)) return 1
-    else if (_.isUndefined(b)) return -1
-    else return a < b ? -1 : 1
-  })
-
-  return minValue
+  if (!_.isUndefined(rangeOrValue.gt) && !_.isUndefined(rangeOrValue.gte)) {
+    return rangeOrValue.gt < rangeOrValue.gte ? rangeOrValue.gt : rangeOrValue.gte
```

Comment:
This `getMinRangeValue` function was recently introduced to restrict returned results depending on their creation date, accordingly to compression/retention log policy for time-series tables.
It is called when the filter option `minValue` is provided.

https://github.com/stelace/stelace/blob/c2609113b4bf4280041c2c5bb62b9ff799a49210/src/services/event.js#L253-L259

Current list endpoint won't use it, except for the event list endpoint.

The purpose of this function is only to get the min value of the filter value or range. The throw is performed outside this function if the value doesn't match this condition:
https://github.com/stelace/stelace/blob/c2609113b4bf4280041c2c5bb62b9ff799a49210/src/util/listQueryBuilder.js#L113

Concerning the merging, it's to handle the special case where `gt` and `gte` is provided at the same time.

Example:
`/events?createdDate[gt]=2020-01-01T00:00:00.000Z&createdDate[gte]=2020-03-05T00:00:00.000Z`
which will be parsed as
```js
createdDate = {
  gt: '2020-01-01T00:00:00.000Z',
  gte: '2020-03-05T00:00:00.000Z'
}

getMinRangeValue(createdDate) === '2020-01-01T00:00:00.000Z'
```

So if the retention limit date is '2020-02-13T00:00:00.000Z', then the request will throw an error because results can only be filtered from '2020-02-13T00:00:00.000Z', not '2020-01-01T00:00:00.000Z'.

This function needs a proper comment on the usage.

Reply:
You’re just partly rephrasing what I said, but you’re not providing an answer.

I compared with all current list endpoints where you can pass `gt[e]` and `lt[e]`. This is why I was talking about inconsistency. Please read my message above again.